mongo/integration/unified/event_verification.go

```diff
@@ -12,28 +12,112 @@ import (
 	"fmt"
 
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/bsontype"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/event"
+)
+
+var (
+	connectionClosedReasons = map[string]string{
+		"stale":      event.ReasonStale,
+		"idle":       event.ReasonIdle,
+		"error":      event.ReasonConnectionErrored,
+		"poolClosed": event.ReasonPoolClosed,
+	}
+
+	checkOutFailedReasons = map[string]string{
+		"poolClosed":      event.ReasonPoolClosed,
+		"timeout":         event.ReasonTimedOut,
+		"connectionError": event.ReasonConnectionErrored,
+	}
 )
 
 type commandMonitoringEvent struct {
 	CommandStartedEvent *struct {
 		Command      bson.Raw `bson:"command"`
 		CommandName  *string  `bson:"commandName"`
 		DatabaseName *string  `bson:"databaseName"`
+		HasServiceID *bool    `bson:"hasServiceId"`
```

Comment:
Should this be `serviceId` based on https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#expectedcommandevent and in other events?

Edit: The spec files use `hasServiceId`. Perhaps this is a spec typo.

Reply:
It should be `hasServiceId`. I'll file a PR to update the spec.

Reply:
@kevinAlbs https://github.com/mongodb/specifications/pull/967